### ISSUE TITLE
2.22.1 to 2.23.1 renamed our mailer queue to mailers

### ISSUE THREAD

**Issue by gingerlime** [rollbar/rollbar-gem#927]
We noticed an odd side-effect that seems to stem from upgrading rollbar from 2.22.1 to 2.23.1

After the upgrade, our jobs that would normally go into the sidekiq `mailer` queue started going to the `mailers` queue... git bisect pinpointed to the rollbar upgrade, but we didn't dig deeper into what change could have caused this.

Curious to know if anyone else experienced this or something similar?

**Comment by waltjones** [comment]
@gingerlime Thank you for the report.

Is this an ActionMailer queue? If so, what version of ActionMailer?

**Comment by gingerlime** [comment]
Hi @waltjones we're using ActionMailer 5.2.3. In this instance we configured Devise to use the mailer.

I realised that we didn't set `config.action_mailer.deliver_later_queue_name` before, but somehow it worked and defaulted to the `mailer` queue. After upgrading rollbar from 2.22.1 to 2.23.1, it started going to the `mailers` queue. I'm a little unsure where did the `mailer` default came from to be honest, and also what changes in the Rollbar gem caused it to switch...

**Comment by waltjones** [comment]
`ActionMailer::DeliveryJob` has had `mailers` as its default queue name for a long time. (It was the default before `ActionMailer::Base.deliver_later_queue_name` was added, as well as after.)

> I realised that we didn't set config.action_mailer.deliver_later_queue_name before, but somehow it worked and defaulted to the mailer queue.

When this is not set by the user, the default is set here: 
https://github.com/rails/rails/blob/master/actionmailer/lib/action_mailer/delivery_methods.rb#L15

The recent PR https://github.com/rollbar/rollbar-gem/pull/915/files loads `ActionMailer::DeliveryJob`, and maybe a change in load order in your app caused the `ActionMailer::DeliveryJob` default to stick. 

Or maybe some other gems were updated at the same time as rollbar-gem.  (Maybe Devise?)

**Comment by gingerlime** [comment]
Thanks @waltjones 

> The recent PR https://github.com/rollbar/rollbar-gem/pull/915/files loads ActionMailer::DeliveryJob, and maybe a change in load order in your app caused the ActionMailer::DeliveryJob default to stick.

Sounds probable. But I'm not aware of anywhere else that was setting the mailer queue to `mailer` instead of the default `mailers`, and that's what so odd about it. I don't know of any other gems that might do this, and I checked Devise and couldn't find any explicit reference to `mailer` (nor `mailers`). I'm really unsure what caused our app to default to `mailer`... In any case, now we've explicitly set it, to avoid relying on anything implicit.

> Or maybe some other gems were updated at the same time as rollbar-gem. (Maybe Devise?)

I isolated the problem to the rollbar gem ugprade alone, eliminating any issues from other gems.

In any case, thanks again for the info. I appreciate your help. I think we can close this now. Hopefully others either use the default queue or set it explicitly (like we now do).
